EurosportDigital/web-toolkit
# (EWL-900): create Input component 📝

# [(EWL-900)](https://eurosportdigital.atlassian.net/browse/EWL-900): create Input component 📝

## Checklist

- Browser compatibility checked:
- [x] MR title respects the format: `(EWL-XXX): description`
- [x] Move Jira ticket into PR

## Design

### Email

|         Mockup         |      New component     |  
| :--------------------: | :--------------------: | 
| ![Screenshot 2019-07-16 at 16 33 04](https://user-images.githubusercontent.com/7658664/61303325-9aecb980-a7e7-11e9-9a4d-95f6ef678ca0.png) | ![Screenshot 2019-07-16 at 16 35 18](https://user-images.githubusercontent.com/7658664/61303410-c2438680-a7e7-11e9-89d1-34ef68ad3bee.png) | 

## Password

|         Mockup         |      New component      |  
| :--------------------: | :--------------------: |
| ![Screenshot 2019-07-16 at 16 36 15](https://user-images.githubusercontent.com/7658664/61303472-d9827400-a7e7-11e9-842d-1e41104b2c21.png) |      ![Screenshot 2019-07-16 at 16 36 42](https://user-images.githubusercontent.com/7658664/61303508-e99a5380-a7e7-11e9-997a-2ef96e765292.png) |

## Example

|         Email         |      Password      |  
| :--------------------: | :--------------------: |
| ![email](https://user-images.githubusercontent.com/7658664/61303572-0c2c6c80-a7e8-11e9-9897-361b63c93fa8.gif) | ![password](https://user-images.githubusercontent.com/7658664/61303618-1a7a8880-a7e8-11e9-97ad-ce467f35dbf3.gif) |

## Description

you can test the Input here: https://deploy-preview-253--es-web-toolkit.netlify.com

- [x] create `input` component
- [x] create two mode: `text` and `password`
- [x] create a `focused` mode to show a different UI
- [x] show / mask `password` feature
- [x] add `eye.svg` icon for the password

## Test Plan

- create an email component
- create an email component controlled with state
- create an password component
- create an password component controlled with state

And then I've try theses four component to see if we match the expected behaviours
